File: backend/buildeR.DAL/Entities/BuildHistory.cs
using buildeR.Common.Enums;
using buildeR.DAL.Entities.Common;
using System;

namespace buildeR.DAL.Entities
{
    public class BuildHistory: Entity
    {
        public int ProjectId { get; set; }
        public int PerformerId { get; set; }
        public BuildStatus BuildStatus { get; set; }
        public int? Duration { get; set; }
        public DateTime? BuildAt { get; set; }
        public DateTime StartedAt { get; set; }
        public string BranchHash { get; set; }
        public string CommitHash { get; set; }
        public int Number { get; set; }

        public virtual User Performer { get; set; }
        public virtual Project Project { get; set; }
    }
}

File: frontend/src/app/shared/models/build-history.ts
import { User } from './user/user';
import { BuildStatus } from './build-status';
import { Project } from './project/project';

export interface BuildHistory {
  id: number;
  project: Project;
  projectId: number;
  number: number;
  performer: User;
  performerId: number;
  duration: number | null;
  buildAt: Date | null;
  startedAt: Date;
  branchHash: string;
  commitHash: string;
  buildStatus: BuildStatus;
}
